src/main/java/com/simple/metrics/kafka/DropwizardReporter.java

```diff
@@ -61,10 +65,14 @@ public void metricRemoval(KafkaMetric kafkaMetric) {
         String name = dropwizardMetricName(kafkaMetric);
         LOGGER.debug("Removing {}", name);
         registry.remove(name);
+        metricNames.remove(name);
     }
 
     @Override
-    public void close() {}
+    public void close() {
+        for (String name: metricNames)
+            registry.remove(name);
```

Comment:
Looks like [MetricRegistry.remove](http://metrics.dropwizard.io/3.1.0/apidocs/com/codahale/metrics/MetricRegistry.html#remove-java.lang.String-) is safe to call again for a metric that's already been removed, so this should be fine.